src/AggregateFunctions/AggregateFunctionGroupArraySorted.cpp

```diff
@@ -0,0 +1,394 @@
+            throw Exception(ErrorCodes::BAD_ARGUMENTS,
+                "Too large limit parameter for groupArraySorted aggregate function, it should not exceed {}",
+                aggregate_function_group_array_sorted_max_element_size);
+    }
+
+    String getName() const override { return "groupArraySorted"; }
+
+    void add(AggregateDataPtr __restrict place, const IColumn ** columns, size_t row_num, Arena * arena) const override
+    {
+        if constexpr (std::is_same_v<T, Field>)
+        {
+            auto row_value = (*columns[0])[row_num];
+            this->data(place).addElement(std::move(row_value), max_elements, arena);
+        }
+        else
+        {
+            auto row_value = assert_cast<const ColumnVector<T> &>(*columns[0]).getData()[row_num];
+            this->data(place).addElement(std::move(row_value), max_elements, arena);
+        }
+    }
+
+    void merge(AggregateDataPtr __restrict place, ConstAggregateDataPtr rhs, Arena * arena) const override
+    {
+        auto & rhs_values = this->data(rhs).values;
+        for (auto rhs_element : rhs_values)
+            this->data(place).addElement(std::move(rhs_element), max_elements, arena);
+    }
+
+    void serialize(ConstAggregateDataPtr __restrict place, WriteBuffer & buf, std::optional<size_t> /* version */) const override
+    {
+        auto & values = this->data(place).values;
+        size_t size = values.size();
+        writeVarUInt(size, buf);
+
+        if constexpr (std::is_same_v<T, Field>)
+        {
+            for (const Field & element : values)
+            {
+                if (element.isNull())
+                {
+                    writeBinary(false, buf);
+                }
+                else
+                {
+                    writeBinary(true, buf);
+                    serialization->serializeBinary(element, buf, {});
+                }
+            }
+        }
+        else
+        {
+            for (const auto & element : values)
+                writeBinaryLittleEndian(element, buf);
+        }
+    }
+
+    void deserialize(AggregateDataPtr __restrict place, ReadBuffer & buf, std::optional<size_t> /* version */, Arena * arena) const override
+    {
+        size_t size = 0;
+        readVarUInt(size, buf);
+
+        if (unlikely(size > max_elements))
+            throw Exception(ErrorCodes::TOO_LARGE_ARRAY_SIZE, "Too large array size, it should not exceed {}", max_elements);
+
+        auto & values = this->data(place).values;
+        values.resize(size, arena);
+
+        if constexpr (std::is_same_v<T, Field>)
+        {
+            for (Field & element : values)
+            {
+                UInt8 is_null = 0;
+                readBinary(is_null, buf);
+                if (!is_null)
+                    serialization->deserializeBinary(element, buf, {});
+            }
+        }
+        else
+        {
+            for (auto & element : values)
```

Comment:
Same as with serializing, batching will be much faster

Reply:
Updated. Also updated `groupArray`.